js/viewer/shareoptions.js

```diff
@@ -350,9 +350,19 @@ var RichdocumentsShareOptions = {
 OC.Plugins.register('OC.Share.ShareDialogView', {
 
 	attach: function (view) {
-
 		RichdocumentsShareOptions.model = view.model;
 
+		if (_.isUndefined(RichdocumentsShareOptions.model)) {
+			throw 'missing OC.Share.ShareItemModel';
```

Comment:
Is this needed? I mean, throwing an exception will likely crash the app (I'm not sure what is the current state, but I don't remember too many "catch" blocks in javascript), which might break things due to some code not being executed.
If this is intended, I don't know the difference between throwing this exception, and letting the app crash with a "RichdocumentsShareOptions.model is undefined" just below.

I assume that the "view.model" can be undefined. Do you have a use case for this? Maybe you can include a short comment in the code.

Reply:
Yes, probably this check is not that useful. Maybe it should just "return" instead and log warning.